File: dcp-solutions/2024-11-28/solution.go
```markdown
package main

import (
    "errors"
)

func checkPossibility(nums []int) bool {
    if len(nums) <= 2 {
        return true
    }

    var count int = 0
    for i := 1; i < len(nums); i++ {
        if nums[i-1] > nums[i] {
            count++
            if count > 1 {
                return false
            }
            if nums[i-1] > nums[i] && (nums[i]+1) <= nums[i-1] {
                continue
            } else if nums[i-1] < nums[i] && (nums[i-1]-1) >= nums[i] {
                continue
            } else {
                return false
            }
        }
    }

    return true
}
```

File: dcp-solutions/2024-11-28/solution.rb
```ruby
def canBeIncreasing(nums)
  prev = Float::INFINITY
  count = 0
  
  nums.each do |num|
    return false if num <= prev
    prev = num if num > prev
    count += 1 unless prev <= num
  end
  
  count < 2
end
```